source/fe/fe_hermite.cc

```diff
@@ -0,0 +1,840 @@
+      hessian_fix.rescale_fe_hermite_values(hessian_fix,
+                                            *this,
+                                            mapping_internal, //_herm,
+                                            output_data.shape_hessians);
+    }
+
+  if ((flags & update_3rd_derivatives) &&
+      (cell_similarity != CellSimilarity::translation))
+    {
+      for (unsigned int k = 0; k < this->n_dofs_per_cell(); ++k)
+        mapping.transform(make_array_view(fe_data.shape_3rd_derivatives, k),
+                          mapping_covariant_hessian,
+                          mapping_internal,
+                          make_array_view(output_data.shape_3rd_derivatives,
+                                          k));
+
+      internal::Rescaler<dim, spacedim, Tensor<3, spacedim>> third_dev_fix;
+      third_dev_fix.rescale_fe_hermite_values(
+        third_dev_fix,
+        *this,
+        mapping_internal, //_herm,
+        output_data.shape_3rd_derivatives);
+    }
+}
+
+
+/*
+ * Special version of function for Hermite elements,
+ * due to special rescaling requirements
+ */
+template <int dim, int spacedim>
+void
+FE_Hermite<dim, spacedim>::fill_fe_face_values(
+  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+  const unsigned int                                          face_no,
+  const hp::QCollection<dim - 1> &                            quadrature,
+  const Mapping<dim, spacedim> &                              mapping,
+  const typename Mapping<dim, spacedim>::InternalDataBase &   mapping_internal,
+  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
+                                                                     spacedim>
+    &,
+  const typename FiniteElement<dim, spacedim>::InternalDataBase &fe_internal,
+  dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim,
+                                                                     spacedim>
+    &output_data) const
+{
+  /*
+   * Convert data object to internal data for this class. Fails with
+   * an exception if that is not possible.
+   */
+  Assert(
+    (dynamic_cast<const typename FE_Hermite<dim, spacedim>::InternalData *>(
+       &fe_internal) != nullptr),
+    ExcInternalError());
+  const typename FE_Hermite<dim, spacedim>::InternalData &fe_data =
+    static_cast<const typename FE_Hermite<dim, spacedim>::InternalData &>(
+      fe_internal);
+
+  Assert(
+    (dynamic_cast<const typename MappingHermite<dim, spacedim>::InternalData *>(
+       &mapping_internal) != nullptr) ||
+      (dynamic_cast<const typename MappingCartesian<dim, spacedim>::InternalData
+                      *>(&mapping_internal) != nullptr),
+    ExcInternalError());
+  /*const typename MappingHermite<dim, spacedim>::InternalData
+     &mapping_internal_herm = static_cast<const typename
+     MappingHermite<dim,spacedim>::InternalData &>( mapping_internal);*/
+
+  AssertDimension(quadrature.size(), 1U);

```

Comment:
```suggestion
  AssertDimension(quadrature.size(), 1);
```
Did you get compiler warning?